contracts/SDUtilityPool.sol

```diff
@@ -0,0 +1,771 @@
+        uint256 rateToPool = (utilizationRatePerBlock * oneMinusProtocolFeeFactor) / DECIMAL;
+        return (poolUtilization() * rateToPool) / DECIMAL;
+    }
+
+    /**
+     * @notice Calculates the exchange rate from the SD to the SDx
+     * @dev This function does not accrue fee before calculating the exchange rate
+     * @return Calculated exchange rate scaled by 1e18
+     */
+    function exchangeRateStored() external view override returns (uint256) {
+        return _exchangeRateStoredInternal();
+    }
+
+    /**
+     * @notice view function to get utilizer latest utilize balance
+     * @param _utilizer address of the utilizer
+     */
+    function getUtilizerLatestBalance(address _utilizer) external view override returns (uint256) {
+        uint256 currentBlockNumber = block.number;
+        uint256 blockDelta = currentBlockNumber - accrualBlockNumber;
+        uint256 simpleFeeFactor = utilizationRatePerBlock * blockDelta;
+        uint256 utilizeIndexNew = (simpleFeeFactor * utilizeIndex) / DECIMAL + utilizeIndex;
+        UtilizerStruct storage utilizeSnapshot = utilizerData[_utilizer];
+
+        if (utilizeSnapshot.principal == 0) {
+            return 0;
+        }
+        uint256 principalTimesIndex = utilizeSnapshot.principal * utilizeIndexNew;
+        return principalTimesIndex / utilizeSnapshot.utilizeIndex;
+    }
+
+    /**
+     * @notice view function to get delegator latest SD balance
+     * @param _delegator address of the delegator
+     */
+    function getDelegatorLatestSDBalance(address _delegator) external view override returns (uint256) {
+        uint256 latestExchangeRate = getLatestExchangeRate();
+        return (latestExchangeRate * delegatorCTokenBalance[_delegator]) / DECIMAL;
+    }
+
+    /**
+     * @notice view function to get latest exchange rate
+     */
+    function getLatestExchangeRate() public view override returns (uint256) {
+        uint256 currentBlockNumber = block.number;
+        uint256 blockDelta = currentBlockNumber - accrualBlockNumber;
+        uint256 simpleFeeFactor = utilizationRatePerBlock * blockDelta;
+        uint256 feeAccumulated = (simpleFeeFactor * totalUtilizedSD) / DECIMAL;
+        uint256 totalUtilizedSDNew = feeAccumulated + totalUtilizedSD;
+        uint256 totalProtocolFeeNew = (protocolFeeFactor * feeAccumulated) / DECIMAL + totalProtocolFee;
+        if (cTokenTotalSupply == 0) {
+            return DECIMAL;
+        } else {
+            uint256 poolBalancePlusUtilizedSDMinusReserves = getPoolAvailableSDBalance() +
+                totalUtilizedSDNew -
+                totalProtocolFeeNew;
+            uint256 exchangeRate = (poolBalancePlusUtilizedSDMinusReserves * DECIMAL) / cTokenTotalSupply;
+            return exchangeRate;
+        }
+    }
+
+    function getPoolAvailableSDBalance() public view override returns (uint256) {
+        return IERC20(staderConfig.getStaderToken()).balanceOf(address(this)) - sdReservedForClaim;
```

Comment:
Does it make it better to add `-sdReservedForWithdrawals` here?

Reply:
Yes otherwise, whenever we require pool available balance, we will have to subtract there.